Sources/Onboarding/VPNNotifications/VPNChurnPromoView.swift

```diff
@@ -0,0 +1,266 @@
+    switch self {
+    case .autoRenewSoonExpire:
+      return Strings.VPN.autoRenewSoonExpirePopOverTitle
+    case .autoRenewDiscount:
+      return Strings.VPN.autoRenewDiscountPopOverTitle
+    case .autoRenewFreeMonth:
+      return Strings.VPN.autoRenewFreeMonthPopOverTitle
+    case .updateBillingSoonExpire:
+      return Strings.VPN.updateBillingSoonExpirePopOverTitle
+    case .updateBillingExpired:
+      return Strings.VPN.updateBillingExpiredPopOverTitle
+    case .subscribeDiscount:
+      return Strings.VPN.subscribeVPNDiscountPopOverTitle
+    case .subscribeVPNProtection:
+      return Strings.VPN.subscribeVPNProtectionPopOverTitle
+    case .subscribeAllDevices:
+      return Strings.VPN.subscribeVPNAllDevicesPopOverTitle
+    }
+  }
+  
+  var description: String? {
+    switch self {
+    case .autoRenewSoonExpire:
+      return Strings.VPN.autoRenewSoonExpirePopOverDescription
+    case .updateBillingSoonExpire:
+      return Strings.VPN.updateBillingSoonExpirePopOverDescription
+    case .updateBillingExpired:
+      return Strings.VPN.updateBillingExpiredPopOverDescription
+    case .subscribeVPNProtection:
+      return Strings.VPN.subscribeVPNProtectionPopOverDescription
+    case .subscribeAllDevices:
+      return Strings.VPN.subscribeVPNAllDevicesPopOverDescription
+    default:
+      return nil
+    }
+  }
+  
+  var subDescription: String? {
+    switch self {
+    case .autoRenewSoonExpire:
+      return Strings.VPN.autoReneSoonExpirePopOverSubDescription
+    case .subscribeVPNProtection, .subscribeAllDevices:
+      return Strings.VPN.subscribeVPNPopOverSubDescription
+    default:
+      return nil
+    }
+  }
+  
+  var buttonTitle: String {
+    switch self {
+    case .autoRenewSoonExpire, .autoRenewDiscount, .autoRenewFreeMonth:
+      return Strings.VPN.autoRenewActionButtonTitle
+    case .updateBillingSoonExpire, .updateBillingExpired:
+      return Strings.VPN.updatePaymentActionButtonTitle
+    case .subscribeDiscount, .subscribeVPNProtection, .subscribeAllDevices:
+      return Strings.VPN.subscribeVPNActionButtonTitle
+    }
+  }
+}
+
+public struct VPNChurnPromoView: View {
+  @Environment(\.presentationMode) @Binding private var presentationMode
+  @State private var height: CGFloat?
+  
+  public var renewAction: (() -> Void)?
+  
+  public var churnPromoType: VPNChurnPromoType
+  
+  private let descriptionItems = [Strings.VPN.checkboxBlockAdsAlternate,
+                          Strings.VPN.popupCheckmarkSecureConnections,
+                          Strings.VPN.checkboxFastAlternate,
+                          Strings.VPN.popupCheckmark247Support]
+  
+  public init(churnPromoType: VPNChurnPromoType) {
+    self.churnPromoType = churnPromoType
+  }
+  
+  public var body: some View {
+    ScrollView {
+      VStack(spacing: 24) {
+        headerView
+        detailView
+          .padding(.bottom, 8)
+        footerView
+      }
+      .background {
+        GeometryReader { proxy in
```

Comment:
This is a trick to adjust scrollview height with respect to VStack for PopOver controller, otherwise there will be unnecessary gap at he bottoms since scrollview has tendency to grow maximum height.

This trick is chosen as the fix here after other alternatives like layout trigger.